pallets/communities/src/lib.rs

```diff
@@ -13,96 +179,202 @@ mod tests;
+		/// This type represents an unique ID for the community
+		type CommunityId: Parameter + MaxEncodedLen;
+
+		/// This type represents a rank for a member in a community
+		type MemberRank: Default + Parameter + MaxEncodedLen;
+
+		/// Type represents interactions between fungibles (i.e. assets)
+		type Assets: fungibles::Inspect<Self::AccountId>
+			+ fungibles::Mutate<Self::AccountId>
+			+ fungibles::Create<Self::AccountId>
+			+ fungibles::Destroy<Self::AccountId>;
+
+		/// Type represents interactions between fungibles (i.e. assets)
+		type Balances: fungible::Inspect<Self::AccountId>
+			+ fungible::Mutate<Self::AccountId>
+			+ fungible::InspectFreeze<Self::AccountId>
+			+ fungible::MutateFreeze<Self::AccountId>;
+
+		/// Because this pallet emits events, it depends on the runtime's
+		/// definition of an event.
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
 		/// Type representing the weight of this pallet
 		type WeightInfo: WeightInfo;
+
+		/// The Communities' pallet id, used for deriving its sovereign account
+		/// ID.
+		#[pallet::constant]
+		type PalletId: Get<frame_support::PalletId>;
+
+		/// The Communities' freeze identifier, used for identifying freezes
+		/// created by the pallet.
+		#[pallet::constant]
+		type FreezeIdentifier: Get<<Self::Balances as fungible::InspectFreeze<Self::AccountId>>::Id>;
+
+		/// Max amount of URLs a community can hold on its metadata.
+		#[pallet::constant]
+		type MaxUrls: Get<u32> + Clone + PartialEq + core::fmt::Debug;
+
+		/// Max amount of locations a community can hold on its metadata.
+		#[pallet::constant]
+		type MaxLocations: Get<u32> + Clone + PartialEq + core::fmt::Debug;
 	}
 
-	// The pallet's runtime storage items.
-	// https://docs.substrate.io/main-docs/build/runtime-storage/
+	/// Store the basic information of the community. If a value exists for a
+	/// specified [`ComumunityId`][`Config::CommunityId`], this means a
+	/// community exists.
+	#[pallet::storage]
+	#[pallet::getter(fn community)]
+	pub(super) type CommunityInfo<T> = StorageMap<_, Blake2_128Concat, CommunityIdOf<T>, Community<T>>;
+
+	/// Store the metadata regarding a community.
 	#[pallet::storage]
-	#[pallet::getter(fn something)]
-	// Learn more about declaring storage items:
-	// https://docs.substrate.io/main-docs/build/runtime-storage/#declaring-storage-items
-	pub type Something<T> = StorageValue<_, u32>;
+	#[pallet::getter(fn metadata)]
+	pub(super) type CommunityMetadata<T: Config> =
+		StorageMap<_, Blake2_128Concat, CommunityIdOf<T>, types::CommunityMetadata<T>>;
+
+	/// Store the list of community members. If some values exist under a
+	/// specified [`ComumunityId`][`Config::CommunityId`] prefix, this means a
+	/// community exists.
+	#[pallet::storage]
+	#[pallet::getter(fn member_rank_for)]
+	pub(super) type CommunityMembers<T> =
+		StorageDoubleMap<_, Blake2_128Concat, CommunityIdOf<T>, Blake2_128Concat, AccountIdOf<T>, MemberRankOf<T>>;
```

Comment:
This storage item could track something a bit more complex than the rank like the `Membership` of each user. I'm also thinking of proposing something like a "passport" in the future where we separately keep track of memberships that would also be transferable.

Reply:
I agree with this. Initially thought about it, but since I lacked a good idea for the format of that structure, ended up deciding for just setting a `MemberRank`. We can always change it to that `MemberPassport`, though.